salesapp.js

```diff
@@ -1,205 +1,105 @@
-  maxCust: 38,
-  avgCookiesPerCust: 2.3,
-  totalDailyCookies: 0,
-  rndmCustPerHour: [],
-  cookiesPerHour: []
-};
-var locAlki = {
-  minCust: 2,
-  maxCust: 16,
-  avgCookiesPerCust: 4.6,
-  totalDailyCookies: 0,
-  rndmCustPerHour: [],
-  cookiesPerHour: []
+function CookieStore(minCustPerHour, maxCustPerHour, avgCookiesPerCust, storeName) {
+  this.storeName = storeName;
+  this.minCustPerHour = minCustPerHour;
+  this.maxCustPerHour = maxCustPerHour;
+  this.avgCookiesPerCust = avgCookiesPerCust;
+  this.totalDailyCookies = 0;
+  this.rndmCustPerHour = [];
+  this.cookiesSoldPerHour = [];
+  cookieStores.push(this);
+  console.dir(this);
 };
 
-//generating random # of customers per hour using min and max inputs and storing that function as a method
-locFirstAndPike.rndmCust = function() {
-  for(var i = 0; i < storeHours.length; i++) {
-    this.rndmCustPerHour[i] = Math.floor(Math.random() * (this.maxCust - this.minCust + 1) + this.minCust);
-  }
-};
-locSeaTacAirport.rndmCust = function() {
-  for(var i = 0; i < storeHours.length; i++) {
-    this.rndmCustPerHour[i] = Math.floor(Math.random() * (this.maxCust - this.minCust + 1) + this.minCust);
-  }
-};
-locSeattleCenter.rndmCust = function() {
-  for(var i = 0; i < storeHours.length; i++) {
-    this.rndmCustPerHour[i] = Math.floor(Math.random() * (this.maxCust - this.minCust + 1) + this.minCust);
-  }
-};
-locCapitalHill.rndmCust = function() {
-  for(var i = 0; i < storeHours.length; i++) {
-    this.rndmCustPerHour[i] = Math.floor(Math.random() * (this.maxCust - this.minCust + 1) + this.minCust);
-  }
-};
-locAlki.rndmCust = function() {
-  for(var i = 0; i < storeHours.length; i++){
-    this.rndmCustPerHour[i] = Math.floor(Math.random() * (this.maxCust - this.minCust + 1) + this.minCust);
-  }
-};
+var locFirstAndPike = new CookieStore(23, 65, 6.3, '1st and Pike');
+var locSeaTacAirport = new CookieStore(3, 24, 1.2, 'SeaTac Airport');
+var locSeattleCenter = new CookieStore(11, 38, 3.7, 'Seattle Center');
+var locCapitalHill = new CookieStore(20, 38, 2.3, 'Capitol Hill');
+var locAlki = new CookieStore(2, 16, 4.6, 'Alki Beach');
 
-//calculating # of cookies sold per hour based on random # of customers and avg cookies sold per customer
-locFirstAndPike.calcCookies = function() {
+CookieStore.prototype.genRndmCust = function() {
   for(var i = 0; i < storeHours.length; i++) {
-    this.cookiesPerHour[i] = Math.floor(this.rndmCustPerHour[i] * this.avgCookiesPerCust);
-  }
-};
-locSeaTacAirport.calcCookies = function() {
-  for(var i = 0; i < storeHours.length; i++) {
-    this.cookiesPerHour[i] = Math.floor(this.rndmCustPerHour[i] * this.avgCookiesPerCust);
-  }
-};
-locSeattleCenter.calcCookies = function() {
-  for(var i = 0; i < storeHours.length; i++){
-    this.cookiesPerHour[i] = Math.floor(this.rndmCustPerHour[i] * this.avgCookiesPerCust);
+    this.rndmCustPerHour[i] = Math.floor(Math.random() * (this.maxCustPerHour - this.minCustPerHour + 1) + this.minCustPerHour);
```

Comment:
This should really be done as `this.rndmCustPerHour.push()`. This way works too, but yeah when you're adding elements to an array you should be using `push()`.